certbot/plugins/dns_common.py

```diff
@@ -0,0 +1,305 @@
+"""Common code for DNS Authenticator Plugins."""
+
+import abc
+import configobj
+import logging
+import os
+import stat
+
+from time import sleep
+
+import zope.interface
+
+from acme import challenges
+
+from certbot import errors
+from certbot import interfaces
+
+from certbot.display import ops
+from certbot.display import util as display_util
+
+from certbot.plugins import common
+
+logger = logging.getLogger(__name__)
+
+
+@zope.interface.implementer(interfaces.IAuthenticator)
+@zope.interface.provider(interfaces.IPluginFactory)
+class DNSAuthenticator(common.Plugin):
+    """Base class for DNS  Authenticators"""
+
+    _attempt_cleanup = False
+
+    @classmethod
+    def add_parser_arguments(cls, add):
+        add('propagation-seconds',
+            default=10,
```

Comment:
For the first plugin where 10 seconds isn't enough, I introduce a pattern for plugins to define their own default ([diff](https://github.com/certbot/certbot/pull/4585/files#diff-e2ab6366a1edcd587391b637b7b58a10R34)). Essentially, 10 seconds because the "default default".